src/ui/components/shared/MaterialIcon.tsx

```diff
@@ -26,6 +43,8 @@ export default function MaterialIcon({
   iconSize = "base",
   ...rest
 }: MaterialIconProps) {
+  useMaterialIconCheck();
```

Comment:
I would probably put this at the `App.tsx` level, so it doesn't start setting timeouts from every single material icon. Also, what do you think about passing in the function that does the timeout setting, but leaving `useEffect` in the component?

Reply:
This only does one check which is why i like co-locating it